src/components/vouchers/VoucherRedeem.tsx

```diff
@@ -25,10 +26,31 @@ export const VoucherRedeem = ({ quota, voucherId }: { quota: TQuota; voucherId:
   const [isChecked, setIsChecked] = useState(false);
   const [quantity, setQuantity] = useState(1);
 
-  // TODO: set voucher availability
+  const [isRedeemedVoucher, setIsRedeemedVoucher] = useState(false);
+  const [isAvailableVoucher, setIsAvailableVoucher] = useState(false);
   const [isExpiredVoucher, setIsExpiredVoucher] = useState(false);
 
-  const { maxPerPerson } = quota;
+  const { availableQuantityForMember = 0, availableQuantity = 0 } = quota;
+
+  useEffect(() => {
+    if (availableQuantity !== 0) {
+      setIsAvailableVoucher(true);
+
+      if (availableQuantityForMember === 0) {
+        setIsRedeemedVoucher(true);
+
+        return;
+      }
+
+      return;
+    }
+
+    if (availableQuantityForMember === 0) {
+      setIsRedeemedVoucher(true);
+
+      return;
```

Comment:
effects should not return anything. returns are used for cleanup functions.
in our case, we do not have the cleanup so there shouldn't be returns. please rework it with ifs and elses

Reply:
chatgpt is refactoring this to

```
useEffect(() => {
  const hasAvailableQuantity = availableQuantity !== 0;
  const hasNoAvailableQuantityForMember = availableQuantityForMember === 0;

  setIsAvailableVoucher(hasAvailableQuantity);
  
  if (hasNoAvailableQuantityForMember) {
    setIsRedeemedVoucher(true);
  }
}, [availableQuantity, availableQuantityForMember]);
```

can you give it a try or find a mixed solution?

Reply:
I tested it and it works. 👍
https://github.com/smart-village-solutions/smart-village-app-app/commit/c6956ae520ab0fe55a1ed53cf90f60bfb9f25bbd